OceanLab/eof.py

```diff
@@ -121,3 +125,154 @@ def my_eof_interp(M,nmodes,errmin=1e-15,repmax=None):
     vi = (M.T+Mmean).T
 
     return vi
+
+#=========================================
+# PERFORM COMPLEX EOF
+#=========================================
+def ceof(lon, lat, data, nkp = 10):
+    ''' Complex (Hilbert) EOF
+    Note: the mean field in each coordinate is subtracted within the function.
+    Do not remove the time-mean before inputing in the fuction.
+    
+    First written in MATLAB and found in Prof. Daniel J. Vimont webpage 
+    (https://www.aos.wisc.edu/~dvimont/matlab/Stat_Tools/complex_eof.html)
+    Version 1.0.0 on 15/Mar/2021
+    ==============================================================================
+    INPUT:
+       lon     = longitudes (array)
+       lat     = latitude (array)
+       data    = original data set [time, lat, lon]
+       nkp     = number of modes to return (default = 10)
+
+    OUTPUT:
+       per     = percent variance explained (real eigenvalues)
+       modes   = first nkp complex loadings or eigenvectors [lat, lon, nkp]
+       SpAmp   = spatial amplitude [lat, lon, nkp]
+       SpPhase = spatial phase [lat, lon, nkp]
+       pcs     = first nkp complex principal components or amplitudes [time, nkp]
+       TAmp    = temporal amplitude [time, nkp]
+       TPhase  = temporal phase [time, nkp]    
+    ==============================================================================
+    Version 2.0.0 on 25/May/2021. 
+        Now, it is possible to input data with NaN values. 
+    ==============================================================================
+    Version 3.0.0 on 17/Jun/2021. 
+        New modifications:
+        (i)  The function organizes the data as time vs space and subtract the mean field in each coordinate
+        (ii) It returns all the variables related to CEOF in a DataArray
+    ''' 
```

Comment:
I'm not sure this version control is useful here.
1) github has version control
2) when we merge these commits, it will be the "actual first version" =P

Reply:
Done